stream-compositions/services/legal-entity-composition-service/src/main/java/com/backbase/stream/compositions/legalentity/core/service/LegalEntityIngestionService.java

```diff
@@ -0,0 +1,92 @@
+package com.backbase.stream.compositions.legalentity.core.service;
+
+import com.backbase.stream.LegalEntitySaga;
+import com.backbase.stream.LegalEntityTask;
+import com.backbase.stream.compositions.legalentity.core.mapper.LegalEntityMapper;
+import com.backbase.stream.compositions.legalentity.core.model.LegalEntityIngestPullRequest;
+import com.backbase.stream.compositions.legalentity.core.model.LegalEntityIngestPushRequest;
+import com.backbase.stream.compositions.legalentity.core.model.LegalEntityIngestResponse;
+import com.backbase.stream.legalentity.model.LegalEntity;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+public class LegalEntityIngestionService {
```

Comment:
maybe we shall use the dependency inversion principle for all the services classes